IBAnimatable/Animatable.swift

```diff
@@ -115,47 +134,50 @@ public extension Animatable where Self: UIView {
   func autoRunAnimation() {
     if autoRun {
       autoRun = false
-      animate()
+      animate(self.animationType)
     }
   }
+}
+
+fileprivate extension Animatable where Self: UIView {
 
   // MARK: - Animation methods
-  public func slide(_ way: AnimationType.Way, direction: AnimationType.Direction, completion: AnimatableCompletion? = nil) {
-    let values = computeValues(way: way, direction: direction, shouldScale: false)
+  func slide(_ way: AnimationType.Way, direction: AnimationType.Direction, configuration: AnimationConfiguration, completion: AnimatableCompletion? = nil) {
```

Comment:
All of them are breaking APIs, right? I don't mind to change them in release 4 and let the user use the new and nice API.